raiden_libs/network/matrix/client.py

```diff
@@ -179,23 +186,34 @@ def typing(self, room: Room, timeout: int=5000):
         return self.api._send('PUT', path, {'typing': True, 'timeout': timeout})
 
     def add_invite_listener(self, callback: Callable):
-        super().add_invite_listener(geventify_callback(callback))
+        super().add_invite_listener(self.geventify(callback))
 
     def add_leave_listener(self, callback: Callable):
-        super().add_leave_listener(geventify_callback(callback))
+        super().add_leave_listener(self.geventify(callback))
 
     def add_presence_listener(self, callback: Callable):
-        return super().add_presence_listener(geventify_callback(callback))
+        return super().add_presence_listener(self.geventify(callback))
 
     def add_listener(self, callback: Callable, event_type: str = None):
-        return super().add_listener(geventify_callback(callback), event_type)
+        return super().add_listener(self.geventify(callback), event_type)
 
     def add_ephemeral_listener(self, callback: Callable, event_type: str = None):
-        return super().add_ephemeral_listener(geventify_callback(callback), event_type)
+        return super().add_ephemeral_listener(self.geventify(callback), event_type)
 
     def _mkroom(self, room_id: str) -> Room:
         """ Uses a geventified Room subclass """
         room = self.rooms[room_id] = Room(self, room_id)
         if not room.canonical_alias:
             room.update_aliases()
         return room
+
+    def join_and_logout(self, greenlets=None, timeout=None):
+        if greenlets is not None:
+            self.greenlets.extend(greenlets)
```

Comment:
It would be a bit nicer to not modify the instance variable but use a copy instead - but not really important